src/multiportgraph/iter.rs

```diff
@@ -8,6 +8,71 @@ use super::{MultiPortGraph, SubportIndex};
 use crate::portgraph::{self, NodePorts};
 use crate::{Direction, LinkView, NodeIndex, PortIndex, PortOffset, PortView};
 
+/// Iterator methods for [`MultiPortGraph`] with concrete return types.
+///
+/// Used internally by other iterator implementations to avoid the generic RPITIT return types.
+impl MultiPortGraph {
+    #[inline]
+    /// Returns an iterator over every pair of matching ports connecting `from`
+    /// with `to`.
+    pub fn _get_connections(&self, from: NodeIndex, to: NodeIndex) -> NodeConnections {
```

Comment:
I didn't think about it since it was in a private module, but they should be `pub(crate)` instead 👍 

I agree that these are annoying to have. In an ideal world we'd be able to use `#[refine]` instead...
https://rust-lang.github.io/rfcs/3245-refined-impls.html